python/datafusion/dataframe.py

```diff
@@ -211,6 +245,19 @@ def sort(self, *exprs: Expr | SortExpr) -> DataFrame:
         exprs_raw = [sort_or_default(expr) for expr in exprs]
         return DataFrame(self.df.sort(*exprs_raw))
 
+    def cast(self, mapping: dict[str, pa.DataType[Any]]) -> DataFrame:
+        """Cast all or a subset of columns to new dtype.
+
+        Args:
+            mapping (dict[str, pa.DataType[Any]]):  Mapped with column as key and column
+                dtype as value.
```

Comment:
Not necessary to have type hints in both the docstring and also the function declaration.